### PR TITLE
Comments parse and enhancement in tests 

### PR DESCRIPTION
**Pull request by Jeconias**
I believe that basic tests are finished. Now, we can advance. @jamillosantos

### CODE REVIEW

File: experiment_test.go [hunk 1]
```diff
@@ -1,19 +1,17 @@
...
-	"github.com/fatih/structtag"
 	myasthurts "github.com/jamillosantos/go-my-ast-hurts"
 	. "github.com/onsi/ginkgo"
 	. "github.com/onsi/gomega"
 )
 
 var _ = Describe("My AST Hurts", func() {
 
-	FContext("should parse struct", func() {
+	Context("should parse struct", func() {
```
Comment (felipemfp) [inline experiment_test.go]: Use `Context` for describing the test context, not what you expect to happen (this is the `It`). For example:

```go
Describe("Categorizing book length", func() {
        Context("With more than 300 pages", func() {
            It("should be a novel", func() {
                Expect(longBook.CategoryByLength()).To(Equal("NOVEL"))
            })
        })

        Context("With fewer than 300 pages", func() {
            It("should be a short story", func() {
                Expect(shortBook.CategoryByLength()).To(Equal("SHORT STORY"))
            })
        })
    })
```

In this case, you could use the `When`:

```suggestion
	When("parsing structs", func() {
```

File: parser.go [hunk 2]
```diff
@@ -43,53 +43,63 @@ func parsePackage(ctx *parseContext) {
...
-			Comment: comment,
+			Comment: comments,
 		}
 		ctx.Env.AppendPackage(pkg)
 		ctx.PackagesMap[ctx.File.Name.Name] = pkg
 	}
 }
 
-func parseComments(doc *ast.CommentGroup) []string {
+func parseComments(doc *ast.CommentGroup, c *[]string) {
```
Comment (felipemfp) [inline parser.go]: Why did you prefer to receive it as a pointer?

**Review by felipemfp — COMMENTED** [review]
Could we have those sample files as `.sample.go`? We would have syntax highlighting and also would ensure we are parsing valid go files (we could have a `go build` in the Workflow).

File: parser.go (hunk 2 above)
Comment (Jeconias) [reply parser.go]: I don't know, but i will go see if this is necessary.

File: experiment_test.go (hunk 1 above)
Comment (Jeconias) [reply experiment_test.go]: Ok, i understand.

File: parser.go (hunk 2 above)
Comment (Jeconias) [reply parser.go]: This has been resolved in branch **builtin-parse**. No is need use pointers.

**Comment by Jeconias** [comment]
> Could we have those sample files as `.sample.go`? We would have syntax highlighting and also would ensure we are parsing valid go files (we could have a `go build` in the Workflow).

Sorry for the delay. Yes, I don't see problem. I believe this will be better.

**Pull request merged**